### ISSUE TITLE
TODO: substitute included static libraries by dynamic linking where conda-forge packages exist

### ISSUE THREAD

**Issue by dlaehnemann** [conda-forge/r-nanoparquet-feedstock#1]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The current build simply takes nanoparquet as is and builds it. This includes a bunch of external dependencies that are simply included in the nanoparquet repo for static linking right here:
https://github.com/r-lib/nanoparquet/tree/main/src

Instead, we want to actually follow [`CFEP 18` to make as many of them as possible dynamic links against other conda-forge packages](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).

This is a tracking issue for this, and I guess a first step will be to determine which dependencies are already on conda-forge and then provide a TODO list here. And we'll need to check which version of them is being used in `nanoparquet`.

### Installed packages

```shell
---
```

### Environment info

```shell
--
```

**Comment by dlaehnemann** [comment]
At the same time, I think having all the dependency libraries bundled in the nanoparquet repo in a working combination is kind of the idea behind this---having a minimum external dependencies package for handling parquet in R . So I guess this is nanoparquet vs. conda-forge philosophy here, and the question whether we can trust this staying in sync with stuff being pulled in by `r-base`, for example...

**Comment by mfansler** [comment]
Agreed, it is a different philosophy. Conda Forge values minimizing redundancy in both the use of CI resources to build software and for the end-user installing compiled code. It also aims for quicker pace of propagating improvements.

In theory, Conda Forge recipes *done correctly* will have the versions of dynamic libraries constrained such that only compatible ABIs are ever installed. In practice, unit tests can also be enabled to more thoroughly ensure functional parity. In reality, there are many oversights leading to end-user troubles and making work for maintainers. The static-everything approach - for example, in the Rust ecosystem - boasts a less painful experience for the end-user.

You're right that one loses some flexibility with what other package *builds* could be co-installed if this package uses Conda Forge dynamic libraries. Only other builds using the same "epoch"<sup>**1**</sup> of dynamic libraries with compatible ABIs could be co-installed.

---

<sub>**[1]** Specifically, the version of `conda-forge-pinning` that defines the ecosystem-wide versions to be used.</sup>

**Comment by dlaehnemann** [comment]
Thanks for completing the migration to `r-base 4.4` for this feedstock!

And do you have a preference / suggestion how we proceed with this, here?

I think I would actually argue for keeping it as is, again. `r-nanoparquet` is really meant as a minimal viable version of `parquet` reading and writing in R, and the minimum dependency philosophy of bundling everything statically makes sense to me in that regard. For more extensive `parquet` functionality, people will probably usually install `r-arrow`, which is much bigger and where the dynamic linking then probably is more important. But if there is a strong preference (and a good argument) for also trying to do things dynamically with `r-nanoparquet`, I could try to put in some effort, here.

**Comment by mfansler** [comment]
I'm not going to push it - there are plenty of more productive things to deal with. The tarball is under 1MB, so it's not like it is some substantial waste of resources. Let's leave this open though, but I'm relabeling as "enhancement" rather than "bug".
